python_ta/checkers/infinite_loop_checker.py

```diff
@@ -0,0 +1,73 @@
+"""Check for infinite while loops."""
+
+from typing import Optional
+
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.lint import PyLinter
+
+
+class InfiniteLoopChecker(BaseChecker):
+    name = "infinite-loop"
+    msgs = {
+        "W9501": (
+            "No condition variable in the while loop is used inside the loop body",
+            "loop-condition-variable-unused",
+            """Used when none of the variables inside a while loop's condition appear in the body. This
+            suggests the condition will never change and may result in an unintended infinite loop.""",
+        ),
+    }
+
+    def __init__(self, linter: Optional[PyLinter] = None) -> None:
+        super().__init__(linter=linter)
+
+    def visit_while(self, node: nodes.While) -> None:
+        self._check_condition_all_var_used(node)
+
+    def leave_while(self, node: nodes.While) -> None:
+        return
+
+    def _check_condition_all_var_used(self, node: nodes.While) -> None:
+        """Helper function that checks whether variables used in a while loop's condition
+        are also used anywhere inside the loop body.
+
+        Note: This is a basic check. It only flags loops where NONE of the condition variables
+        appear in the body, which indicates an infinite loop.
+        """
+        # Get variable(s) used inside condition
+        cond_vars = set()
+        for child in node.test.nodes_of_class((nodes.Name, nodes.Attribute)):
```

Comment:
Considering attributes isn't a bad idea in general, but it is more complex because attributes can be modified through method calls, so it isn't enough to check for whether the attribute is used directly in the loop body.

So, you should remove checks for attributes, and instead just focus on variable names (i.e., `Name` nodes). Moreover, you don't need to exclude `self`; if `self` appears in the loop condition, e.g. through attribute access or a method call, it's valid to also check whether `self` is used in the loop body.